### PR TITLE
fix: avoid browser hanging on close window while video is recording (closes #6037)

### LINKED ISSUE #6037: Testcase hangs after expecting in two different windows. Testcafe V1.12 Chrome 89.
### What is your Test Scenario?
Running a test that requires interaction with two windows and expecting in two windows.
 
### What is the Current behavior?
The tests hangs if expecting in both windows with Chrome.
 
### What is the Expected behavior?
Finish the test result.
 
### What is your web application and your TestCafe test code?
Running testcafe on website that requires access. However I replicated this issue on a public website. 

The issue comes when expecting in two different windows. If the new created window did not had any expect the tests result would finish. However if both windows are expected the tests hangs when expecting in the main or parent window.

 
```
fixture('Two windows - expect bug')
    .page('https://www.dr.dk/');
test
('Open website page in 2 windows and expect', async t => {
    await t
        .click(Selector('button[name="submitAll"]'))
        .click(Selector('button[class="dre-search-input__button dre-search-input--expand__button"]'))
        .typeText(Selector('input[name="query"]'), 'foo')
        .openWindow('https://www.dr.dk/').resizeWindow(1200,700)
        .typeText(Selector('input[name="query"]'), 'foo')
        .pressKey('enter')
        .expect(Selector('a[href="https://www.dr.dk/"]').exists).ok({ timeout: 5000 })
        .closeWindow()
        .pressKey('enter')
        .expect(Selector('a[href="https://www.dr.dk/"]').exists).ok({ timeout: 5000 })
        .expect(Selector('span').withText('Yuk Foo | Wolf Alice').exists).ok({ timeout: 5000 });
}).only;
 
```
### Your Environment details:
 
* testcafe version:                   1.12.0
* node.js version:                    14.15.4
* command-line arguments:  testcafe chrome src/tests -c 1
* browser name and version: Chrome v89
* platform and version:          Windows 10

### PR DESCRIPTION
**Pull request by AlexKamaev**
Closes #6037

### CODE REVIEW

File: src/browser/connection/gateway.ts [hunk 1]
```diff
@@ -184,6 +186,15 @@ export default class BrowserConnectionGateway {
         }
     }
 
+    private static _onCloseWindowRequest (req: IncomingMessage, res: ServerResponse, connection: BrowserConnection): void {
+        if (BrowserConnectionGateway._ensureConnectionReady(res, connection)) {
+            connection.provider.closeBrowserChildWindow(connection.id)
```
Comment (miherlosev) [inline src/browser/connection/gateway.ts:191]: Should we catch errors of the `connection.provider.closeBrowserChildWindow` method here?

File: src/browser/provider/built-in/dedicated/chrome/cdp-client/index.ts [hunk 2]
```diff
@@ -385,4 +415,11 @@ export class BrowserClient {
     public switchToMainWindow (): void {
         ExecutionContext.switchToMainWindow();
     }
+
+    public async closeBrowserChildWindow (): Promise<void> {
+        await this.setClientInactive();
+
+        // NOTE: delay browser window closing
+        await delay(100);
```
Comment (miherlosev) [inline src/browser/provider/built-in/dedicated/chrome/cdp-client/index.ts:423]: Can we use here a more stable way to check that window is closed?

File: src/browser/provider/plugin-host.js [hunk 3]
```diff
@@ -153,4 +153,8 @@ export default class BrowserProviderPluginHost {
     getConfig (value) {
         return value;
     }
+
+    closeBrowserChildWindow (/*browserId*/) {
```
Comment (miherlosev) [inline src/browser/provider/plugin-host.js:157]: `async closeBrowserChildWindow (/*browserId*/) {}`

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**